tests/platform_tests/api/test_component.py

```diff
@@ -48,6 +58,25 @@ def setup(self, platform_api_conn):
             except:
                 pytest.fail("num_components is not an integer")
 
+    #
+    # Helper functions
+    #
+
+    def compare_value_with_platform_facts(self, key, value, component_idx):
+        expected_value = None
+
+        if self.chassis_facts:
+            expected_components = self.chassis_facts.get("components")
+            if expected_components:
+                expected_value = expected_components[component_idx].get(key)
+
+        if not expected_value:
+            logger.warning("Unable to get expected value for '{}' from platform.json file for component {}".format(key, component_idx))
+            return
```

Comment:
Remove warning message and make the check mandatory. See SFP portion of https://github.com/Azure/sonic-mgmt/pull/2268 for examples.